File: TUF/Array/SortElmByFrequency.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function sortElementByFrequency(arr) {
    if (arr.length === 0) {
        console.log("Array is empty");
        return;
    }
    var map = new Map();
    var newArr = [];
    for (var i = 0; i < arr.length; i++) {
        map.set(arr[i], (map.get(arr[i]) || 0) + 1);
    }
    while (map.size !== 0) {
        var maxElm = findMaxFrequency(map);
        var maxFreq = map.get(maxElm);
        for (var j = 0; j < maxFreq; j++) {
            newArr.push(maxElm);
        }
        map.delete(maxElm);
    }
    console.log(newArr.join(", "));
}
function findMaxFrequency(map) {
    var maxElm = -1;
    var maxFreq = 0;
    for (var _i = 0, _a = map.entries(); _i < _a.length; _i++) {
        var _b = _a[_i], key = _b[0], value = _b[1];
        if (value > maxFreq) {
            maxElm = key;
            maxFreq = value;
        }
    }
    return maxElm;
}
var arr = [1, 2, 3, 2, 4, 3, 1, 2];
sortElementByFrequency(arr);

File: TUF/Array/SortElmByFrequency.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class SortElmByFrequency {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        input.close();
        sortElementByFrequency(arr);
    }

    private static void sortElementByFrequency(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] newArr = new int[arr.length];
        int count = 0;
        for (int i = 0; i < arr.length; i++) {
            map.put(arr[i], (map.getOrDefault(arr[i], 0) + 1));
        }
        while (!map.isEmpty()) {
            int maxElm = findMaxFrequency(map);
            int maxFreq = map.get(maxElm);
            for (int j = 0; j < maxFreq; j++) {
                newArr[count] = maxElm;
                count++;
            }
            map.remove(maxElm);
        }
        for (int i = 0; i < newArr.length; i++) {
            System.out.print(newArr[i] + ", ");
        }
    }

    private static int findMaxFrequency(HashMap<Integer, Integer> map) {
        int maxFreq = 0;
        int maxElm = -1;
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() > maxFreq) {
                maxFreq = entry.getValue();
                maxElm = entry.getKey();
            }
        }
        return maxElm;
    }
}
